projects/optic/src/commands/oas/captures/streams/sources/proxy.ts

```diff
@@ -37,14 +37,17 @@ export class ProxyInteractions {
     options: {
       ca?: ProxyCertAuthority;
       targetCA?: Array<{ cert: Buffer | string }>;
-    } = {}
+      mode: 'reverse-proxy' | 'system-proxy';
+    }
   ): Promise<[ProxyInteractions, string, string]> {
-    let { host, protocol } = new URL(targetHost);
+    let { host, protocol, origin } = new URL(targetHost);
     if (targetHost.includes('/')) {
       // accept urls to be passed in rather than pure hosts
       targetHost = host;
     }
 
+    const forwardHost = options.mode === 'reverse-proxy' ? origin : targetHost;
```

Comment:
Something changed downstream in mockttp that we need different hosts to forward to in 

Takes into account:
- reverse proxy needs the protocol as the forward host 
- system proxy needs the host (https://github.com/opticdev/optic/pull/1888 - fixed regression here)

I'm not entirely sure how this all fits together but I was able to hit github's api via this

Reply:
Interesting. Might be good to document this in the code too -- I wonder if it may change again